File: nodejs/src/muse/algorithms/number_theory/Coprimality.js
"use strict"

const GreatestCommonDivisor = require("muse/algorithms/number_theory/GreatestCommonDivisor")

exports.reduceToBinaryGCD = (m, n) => {
    return GreatestCommonDivisor.iterativeBinaryGCD(m, n) === 1
}

exports.reduceToEuclidean = (m, n) => {
    return GreatestCommonDivisor.iterativeEuclidean(m, n) === 1
}

File: python/muse/algorithms/number_theory/coprimality.py
from muse.algorithms.number_theory import greatest_common_divisor


def reduce_to_binary_gcd(m: int, n: int) -> bool:
    return greatest_common_divisor.iterative_binary_gcd(m, n) == 1


def reduce_to_euclidean(m: int, n: int) -> bool:
    return greatest_common_divisor.iterative_euclidean(m, n) == 1
